hypervisor/arch/x86/guest/vmsr.c

```diff
@@ -262,34 +262,34 @@ uint32_t vmsr_get_guest_msr_index(uint32_t msr)
 	return index;
 }
 
-static void enable_msr_interception(uint8_t *bitmap, uint32_t msr_arg, uint32_t mode)
+static void enable_msr_interception(uint8_t bitmap[], uint32_t msr_arg, uint32_t mode)
```

Comment:
using [] for pointer is very rare... What is the policy?  Can you ask Xiangyang to clarify?

Reply:
@wuxyintel

Reply:
While waiting for Xiangyang's comment ... before this change, I checked the C language and I believe it asks compiler to treat it as pointer: "A declaration of a parameter as ‘‘array of type’’ shall be adjusted to ‘‘qualified pointer to type’’, where the type qualifiers (if any) are those specified within the [ and ] of the array type derivation.". I compared the assembly code of bitmap[] and *bitmap and it's identical.

Reply:
Hi Eddie and Zide, from MISRA C view, "bitmap[]" is variable-length array, this is violation, we shall avoid to use variable-length array. 
Rational: Parameter "bitmap[]" will be stored on the stack, it is difficult to determine statically how many memory shall be reserved for a stack. This will violate the principle of "static resource allocation".
BTW, It is compiler's specific behavior that the assembly code of bitmap[] and *bitmap and it's identical. If we make a deviation about this violation, we should add effort to verify this behavior during compiler verification process.
Why we don't use "const struct acrn_vm *vm" instead of "bitmap" in this case, I remember Eddie suggest it.